### ISSUE TITLE
Could not load file or assembly 'System.Net.Http ...

### ISSUE THREAD

**Issue by Sashank999** [GitCredentialManager/git-credential-manager#474]
**Which version of GCM Core are you using?**

Getting the below attached error.

**Which Git host provider are you trying to connect to?**

* [ ] Azure DevOps
* [ ] Azure DevOps Server (TFS/on-prem)
* [x] GitHub
* [ ] GitHub Enterprise
* [ ] Bitbucket
* [ ] Other - please describe

**Can you access the remote repository directly in the browser using the remote URL?**

* [x] Yes
* [ ] No, I get a permission error
* [ ] No, for a different reason - please describe

**Expected behavior**

Able to push my local branch to my remote branch.

**Actual behavior**

Cannot push to my repo. Everything worked fine just a day before. IDK what happened but I can't push to my remote repo.
I recently changed from Password to Github Personal Access Token and even then it worked fine.
But suddenly, Git GUI stopped working and asked me Github username and password.
I tried running push in console but no luck.
I am just getting the below error.
I am running Git for Windows 2.33.0.2. I tried reinstalling but it didn't work.

**Logs**

I have got the following output:
```
~/Desktop/Git Servers/torntools_extension (torntools_extension|abroadFilterFix) $ git push fork abroadFilterFix
12:03:54.460946 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
12:03:54.467091 git.c:455               trace: built-in: git push fork abroadFilterFix
12:03:54.500878 run-command.c:666       trace: run_command: git remote-https fork https://github.com/Sashank999/torntools_extension
12:03:54.524839 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:03:54.527107 git.c:744               trace: exec: git-remote-https fork https://github.com/Sashank999/torntools_extension
12:03:54.527107 run-command.c:666       trace: run_command: git-remote-https fork https://github.com/Sashank999/torntools_extension
12:03:54.559866 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:03:55.164535 run-command.c:666       trace: run_command: 'git credential-manager-core get'
12:03:55.356773 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:03:55.365220 git.c:744               trace: exec: git-credential-manager-core get
12:03:55.365220 run-command.c:666       trace: run_command: git-credential-manager-core get

Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   --- End of inner exception stack trace ---
   at Microsoft.Git.CredentialManager.Program.Main(String[] args)
12:03:55.559038 run-command.c:666       trace: run_command: 'C:/Program Files/Git/mingw64/libexec/git-core/git-gui--askpass' 'Username for '\''https://github.com'\'': '
error: unable to read askpass response from 'C:/Program Files/Git/mingw64/libexec/git-core/git-gui--askpass'
12:03:57.229789 run-command.c:666       trace: run_command: bash -c 'cat >/dev/tty && read -r line </dev/tty && echo "$line"'
Username for 'https://github.com':
12:03:59.282103 run-command.c:666       trace: run_command: 'C:/Program Files/Git/mingw64/libexec/git-core/git-gui--askpass' 'Password for '\''https://github.com'\'': '
error: unable to read askpass response from 'C:/Program Files/Git/mingw64/libexec/git-core/git-gui--askpass'
12:04:00.816170 run-command.c:666       trace: run_command: bash -c 'cat >/dev/tty && read -r -s line </dev/tty && echo "$line" && echo >/dev/tty'
Password for 'https://github.com':
12:04:02.304797 run-command.c:666       trace: run_command: 'git credential-manager-core erase'
12:04:02.490915 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:04:02.496254 git.c:744               trace: exec: git-credential-manager-core erase
12:04:02.496254 run-command.c:666       trace: run_command: git-credential-manager-core erase

Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   --- End of inner exception stack trace ---
   at Microsoft.Git.CredentialManager.Program.Main(String[] args)
remote: No anonymous write access.
fatal: Authentication failed for 'https://github.com/Sashank999/torntools_extension/'
12:04:02.865687 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
12:04:02.870199 git.c:455               trace: built-in: git config --get remote.origin.url
```

**Comment by mjcheetham** [comment]
@Sashank999 can you please answer the following questions so that I can debug further? Thanks!

1. What version of Windows are you running? Please run `ver` inside of command prompt and reply with the result.
2. Are you using Git from the command line, or from inside an IDE (like Visual Studio) or Git GUI?
3. What version of .NET Framework do you have installed? See documentation [here](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed) to work out which versions you have.

**Comment by Sashank999** [comment]
> @Sashank999 can you please answer the following questions so that I can debug further? Thanks!
> 
> 1. What version of Windows are you running? Please run `ver` inside of command prompt and reply with the result.
> 2. Are you using Git from the command line, or from inside an IDE (like Visual Studio) or Git GUI?
> 3. What version of .NET Framework do you have installed? See documentation [here](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed) to work out which versions you have.

1. ```Microsoft Windows [Version 10.0.19043.1237]```

2. I use it from both command line and Git GUI. I use Notepad++ but it doesn't have any Git integrations.

3. I got this output with https://github.com/jmalarcon/DotNetVersions
```Currently installed "classic" .NET Versions in the system:
4.0.0.0
4.8.04084```

**Comment by DaMoochie8trillion** [comment]
I'm sorry I will figure out and see what's going on. Thanks and sorry!

**Comment by mjcheetham** [comment]
@sashank999 there seems to be something strange with your .NET Framework installation. Not sure why the runtime would not be able to find this assembly.

Please can you try installing the latest .NET Framework, or repairing it:

https://dotnet.microsoft.com/download/dotnet-framework/net48

**Comment by Sashank999** [comment]
> @Sashank999 there seems to be something strange with your .NET Framework installation. Not sure why the runtime would not be able to find this assembly.
> 
> Please can you try installing the latest .NET Framework, or repairing it:
> 
> https://dotnet.microsoft.com/download/dotnet-framework/net48

I'm really sorry I forgot to reply with the solution.
I tried repairing it from the Control Panel but it didn't work. I remember downloading the setup for another version of .NET framework but it didn't work and showed that a newer version was already installed.

I have found some C# tutorials and some documentation online and found that my framework is missing "System.Net.Http" package in the GAC.

- I downloaded the latest version of it from NuGet: https://www.nuget.org/packages/System.Net.Http/
- Extracted it with 7-Zip
- Installed the DLL via PowerShell into GAC, instructions in this Stack Overflow answer: https://stackoverflow.com/a/69646898

I don't know how long it will work, and I know that I should be using the version of package that has the assembly version `4.0.0.0` based on the logs, but I took the risk and its been working for a month and I have found no issues.
